tests/ir/in10.re

```diff
@@ -0,0 +1 @@
+^$|^$
```

Comment:
I don't think this is actually testing the change properly -- running `build/bin/re -p -rpcre -lirjson '^$|^$'` on `main` still results in 
```
{
	"start": 0,
	"states": [
		{
			"end": true,
			"strategy": "none",
		}
	]
}
```
which is the same as the test below, give or take the trailing comma after `"strategy": "none"`.

Do we have example inputs `main` does not accept, but should, and the change fixes it? I thought that e.g. `/^abcde$|/` was one of them, but when I try to check that I get an assertion failure:
```
$ build/bin/re -p -rpcre '^abcde$|' bananas
re: src/re/main.c:383: void carryopaque(const struct fsm *, const fsm_state_t *, size_t, struct fsm *, fsm_state_t): Assertion `fsm_getopaque(src_fsm, src_set[i]) != NULL' failed.
Aborted (core dumped)
```
That may be an unrelated issue though.

Reply:
The assert seems to fail both on this branch and `main`, and the simplest input that triggers it seems to be `build/bin/re -p -rpcre 'a|' a`. The `|` and `-p` are both necessary.

Reply:
> I don't think this is actually testing the change properly -- running build/bin/re -p -rpcre -lirjson '^$|^$' on main still results in

that's right - we always did produce the IR_NONE node, and that isn't changing by this PR. I wanted to add it to the corpus just to be sure we do produce IR_NONE for that situation.

What's changing here is how IR_NONE gets converted to VM opcodes (we used to always construct VM_END_FAIL, and now that's done conditionally), and we don't have test coverage for any of the VM opcodes.

Reply:
I think the assert failure is happening because of clashing assumptions between glushkovise, which sets the state as an end (https://github.com/katef/libfsm/blob/main/src/libfsm/glushkovise.c#L144) but some of those states may not have previously been end states and haven't had their opaques copied over yet (https://github.com/katef/libfsm/blob/main/src/libfsm/glushkovise.c#L151-L152). The `carryopaque` hook for `re` asserts that their opaque is non-NULL (https://github.com/katef/libfsm/blob/main/src/re/main.c#L383), so when fsm_glushkovise uses the callback to copy them over, the assert fails. (If I comment out the assert, it leads to the result I'd expect, complaining about ambiguity.) It seems like determinising an NFA created by unioning FSMs whose start state also has an end state in their epsilon closure triggers the assert.

Given some other edge cases I ran into with `fsm_glushkovise` (which is mutating the FSM in place, and may temporarily break various implicit invariants for opaques), this sort of issue isn't that surprising, but it may be really tricky to fix. The end ID changes should obsolete the getopaques mechanism (and already accounts for `fsm_glushkovise`), so we can probably just address this by removing them soon?

Reply:
It also means the assert failure should be specific to the `re` CLI, unless other carryopaque hooks have the same assert.